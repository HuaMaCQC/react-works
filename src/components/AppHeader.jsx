import React from 'react';
import { Label, Icon } from 'semantic-ui-react';
import styles from './AppHeader.styledJsx.scss';

function resolveScopedStyles(scope) {
  return scope.props.className;
}

export default function AppHeader(props) {
  const { switchMenu, user } = props;
  const jsxId = resolveScopedStyles(
    <scope>
      <style jsx>{styles}</style>
    </scope>,
  );

  return (
    <div className="header">
      <div className="menu-icon">
        <Icon data-testid="switchMenu" link name="sidebar" size="big" onClick={switchMenu} />
      </div>
      <span className="title">管理系統</span>
      <div className="user">
        <div className="notify-bar">
          <Label as="a" size="mini" className={`${jsxId} header-bnt`}>
            <Icon size="large" name="envelope" />
            信件
            <Label circular color="red" className={`${jsxId} user-floating`} floating>
              22
            </Label>
          </Label>
          <Label as="a" size="mini" className={`${jsxId} header-bnt`}>
            <Icon size="large" name="comment" />
            訊息
            <Label circular color="red" className={`${jsxId} user-floating`} floating>
              6
            </Label>
          </Label>
        </div>
        <span className="user-title">歡迎:</span>
        <Label as="a" size="large" className={`${jsxId} header-bnt`}>
          <Icon size="large" name="user circle" />
          {user.name}
        </Label>
      </div>
      <style jsx="true">{styles}</style>
    </div>
  );
}
