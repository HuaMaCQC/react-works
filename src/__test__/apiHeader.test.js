/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
} from '@testing-library/react';
import 'jest-dom/extend-expect';
import AppHeader from '../components/AppHeader';

describe('search UI', () => {
  const mock = {
    user: { id: 1, name: 'HuaMa' },
    switchMenu: jest.fn(),
  };

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it('Snapshot Testing', () => {
    const { container } = render(<AppHeader {...mock} />);

    expect(container).toMatchSnapshot();
  });

  it('switchMenu', () => {
    const {
      getByTestId,
    } = render(<AppHeader {...mock} />);

    const switchMenuBtn = getByTestId('switchMenu');

    fireEvent.click(switchMenuBtn);

    expect(mock.switchMenu).toHaveBeenCalledTimes(1);
  });
});
