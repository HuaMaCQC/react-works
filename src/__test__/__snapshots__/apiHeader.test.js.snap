// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search UI Snapshot Testing 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="menu-icon"
    >
      <i
        aria-hidden="true"
        class="sidebar big link icon"
        data-testid="switchMenu"
      />
    </div>
    <span
      class="title"
    >
      管理系統
    </span>
    <div
      class="user"
    >
      <div
        class="notify-bar"
      >
        <a
          class="ui mini label undefined header-bnt"
        >
          <i
            aria-hidden="true"
            class="envelope large icon"
          />
          信件
          <div
            class="ui red circular floating label undefined user-floating"
          >
            22
          </div>
        </a>
        <a
          class="ui mini label undefined header-bnt"
        >
          <i
            aria-hidden="true"
            class="comment large icon"
          />
          訊息
          <div
            class="ui red circular floating label undefined user-floating"
          >
            6
          </div>
        </a>
      </div>
      <span
        class="user-title"
      >
        歡迎:
      </span>
      <a
        class="ui large label undefined header-bnt"
      >
        <i
          aria-hidden="true"
          class="user circle large icon"
        />
        HuaMa
      </a>
    </div>
    <style
      jsx="true"
    >
      AppHeader.styledJsx.scss
    </style>
  </div>
</div>
`;
